File: src/Entity/Restaurant.php
<?php

namespace App\Entity;

use App\Repository\RestaurantRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
 

/**
 * @ORM\Entity(repositoryClass=RestaurantRepository::class)
 */
class Restaurant
{
   
    /**
     * @ORM\Id
     * @ORM\Column(type="string", length=255)
     * *@Assert\NotBlank(message="reference is required")
     */
    private $referResto;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="nom du restaurant is required")
     */
    private $nomResto;

    /**
     * @ORM\Column(type="string", length=255)
     * *@Assert\NotBlank(message="adresse is required")
     */
    private $adresseResto;

    /**
     * @ORM\Column(type="string", length=255)
     * *@Assert\NotBlank(message="telephone is required")
     *  @Assert\Length(
     *      min = 8,
     *      max = 8,
     *      minMessage = "Le numéro doit avoir au minimum {{ limit }} caractères"
     * )
     */
    private $tel;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="nombre d'etoile is required")
     */
    private $nbEtoile;

    /**
     * @ORM\ManyToOne(targetEntity=Categorie::class, inversedBy="Restaurant")
     * @Assert\NotBlank(message="categorie is required")
     */
    private $Categorie;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getReferResto(): ?string
    {
        return $this->referResto;
    }

    public function setReferResto(string $referResto): self
    {
        $this->referResto = $referResto;

        return $this;
    }

    public function getNomResto(): ?string
    {
        return $this->nomResto;
    }

    public function setNomResto(string $nomResto): self
    {
        $this->nomResto = $nomResto;

        return $this;
    }

    public function getAdresseResto(): ?string
    {
        return $this->adresseResto;
    }

    public function setAdresseResto(string $adresseResto): self
    {
        $this->adresseResto = $adresseResto;

        return $this;
    }

    public function getTel(): ?string
    {
        return $this->tel;
    }

    public function setTel(string $tel): self
    {
        $this->tel = $tel;

        return $this;
    }

    public function getNbEtoile(): ?string
    {
        return $this->nbEtoile;
    }

    public function setNbEtoile(string $nbEtoile): self
    {
        $this->nbEtoile = $nbEtoile;

        return $this;
    }

    public function getCategorie(): ?Categorie
    {
        return $this->Categorie;
    }

    public function setCategorie(?Categorie $Categorie): self
    {
        $this->Categorie = $Categorie;

        return $this;
    }
}

File: src/trood/entity/Restaurant.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package trood.entity;

/**
 *
 * @author Rania
 */
public class Restaurant {
    private String refer_resto;
    private String nom_resto;
    private String adresse_resto;
    private String tel;
    private String nb_etoile;
    private String id_categorie;

    public Restaurant(String refer_resto, String nom_resto, String adresse_resto, String tel, String nb_etoile, String id_categorie) {
        this.refer_resto = refer_resto;
        this.nom_resto = nom_resto;
        this.adresse_resto = adresse_resto;
        this.tel = tel;
        this.nb_etoile = nb_etoile;
        this.id_categorie = id_categorie;
    }

 

    public String getRefer_resto() {
        return refer_resto;
    }

    public void setRefer_resto(String refer_resto) {
        this.refer_resto = refer_resto;
    }

    public String getNom_resto() {
        return nom_resto;
    }

    public void setNom_resto(String nom_resto) {
        this.nom_resto = nom_resto;
    }

    public String getAdresse_resto() {
        return adresse_resto;
    }

    public void setAdresse_resto(String adresse_resto) {
        this.adresse_resto = adresse_resto;
    }

    public String getTel() {
        return tel;
    }

    public void setTel(String tel) {
        this.tel = tel;
    }

    public String getNb_etoile() {
        return nb_etoile;
    }

    public void setNb_etoile(String nb_etoile) {
        this.nb_etoile = nb_etoile;
    }

    @Override
    public String toString() {
        return "Restaurant{" + "refer_resto=" + refer_resto + ", nom_resto=" + nom_resto + ", adresse_resto=" + adresse_resto + ", tel=" + tel + ", nb_etoile=" + nb_etoile + '}';
    }

    public String getId_categorie() {
        return id_categorie;
    }

    public void setId_categorie(String Id_categorie) {
        this.id_categorie = Id_categorie;
    }
    
    
				

				

			

			

	
    
}
